package.json

```diff
@@ -37,20 +37,35 @@
   },
   "devDependencies": {
     "@babel/preset-react": "^7.18.6",
+    "@juggle/resize-observer": "^3.4.0",
     "@rollup/plugin-typescript": "^10.0.1",
     "@strapi/eslint-config": "^0.1.2",
+    "@swc/core": "^1.3.11",
+    "@swc/jest": "^0.2.23",
+    "@testing-library/jest-dom": "5.16.5",
+    "@testing-library/react": "12.1.4",
+    "@testing-library/user-event": "^14.4.3",
     "@tsconfig/node16": "^1.0.3",
+    "@types/jest": "^29.2.6",
+    "@types/react": "17.0.2",
+    "@types/react-dom": "17.0.2",
     "@vitejs/plugin-react": "^2.2.0",
     "chokidar-cli": "^3.0.0",
     "cross-env": "^7.0.3",
     "eslint": "^8.28.0",
     "eslint-plugin-mdx": "^1.17.0",
     "fs-extra": "^10.0.0",
     "husky": "=4",
+    "intl": "^1.2.5",
+    "jest": "^29.2.2",
+    "jest-environment-jsdom": "^29.2.2",
+    "jest-styled-components": "^7.1.1",
     "lerna": "^4.0.0",
     "lint-staged": ">=10",
     "plop": "^2.7.4",
     "prettier": "^2.8.0",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
```

Comment:
Mh I don't think we should to that - only the packages that have a dependency on react should declare it. Why do want to have this here? What kind of dev conflicts are you running in?

> react to both utilise workspaces (avoid dev conflicts) and mirror similar structure to the strapi/strapi repo

In the stapi repo we also wanted to move it out of the global dependencies, because there it makes even less sense (the BE code doesn't know about react e.g.).

Reply:
Because it's a dev dependency and my POV is there's no point declaring them per package when we're using workspaces, just like with jest.

Otherwise you have to update the dep in more than one place seems more of a chore imo.

I've seen it used in common practice on numerous repos - and it'll things such as "using more than one version of X" that crops up.

Strapi it makes sense but I think that's not comparable because of the blend of FS code.